crates/walrus-sui/src/client/retry_client.rs

```diff
@@ -1202,6 +1202,19 @@ impl From<tonic::Status> for RetriableClientError {
     }
 }
 
+impl RetriableClientError {
+    fn is_eligible_for_fallback(&self, next_checkpoint: u64) -> bool {
+        match self {
+            Self::RpcError(rpc_error) if rpc_error.status.code() == tonic::Code::NotFound => {
+                rpc_error
+                    .checkpoint_seq_num
+                    .is_some_and(|height| next_checkpoint < height)
+            }
+            _ => true,
```

Comment:
Question: should default be `false`?

Reply:
In case of  missing events error, the error code is `InternalError`, so it will go to fallback

Reply:
@halfprice yes, we want to fallback for all errors except for most recent checkpoints (since we are not sure if they have been produced or not yet)